### PR TITLE
REST API - File Upload & Download

### PR DESCRIPTION
**Pull request by mariodavid**
This PR contains:

* File Uploading
* Different approaches to upload files (multipart/form-data vs. regular content-type)
* File Downloading
* extraction of "Manage Files" to a dedicated section within the REST API docs
* new example on how to reference a file to an entity

✓ Grammarly

What has been not taken over is the HTML example from the CUBA docs (https://doc.cuba-platform.com/restapi-7.1/#rest_api_v2_ex_file_upload) on uploading a file via a form.

### CODE REVIEW

File: content/modules/rest/pages/managing-files.adoc [hunk 1]
```diff
@@ -0,0 +1,174 @@
+= Manage Files
```
Comment (knstvk) [inline content/modules/rest/pages/managing-files.adoc]: What about naming all sections about generic endpoints "... API"?
Entities API
Files API
...

File: content/modules/rest/pages/managing-files.adoc [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+== Uploading Files
+
+The Upload Files API endpoint `/files` used with a `POST` HTTP method allows you to transfer a file to the Jmix application.
+
+There are two different ways of transferring the data:
+
+1. Using `multipart/form-data`
+2. Using different content types with the body containing the binary data
+
+Depending on the use-case either of those two options might be a better fit. In both situations, the API returns `201 - Created` to indicate that the file was successfully stored.
```
Comment (knstvk) [inline content/modules/rest/pages/managing-files.adoc]: "use case"

**Review by knstvk — APPROVED** [review]
Looks good!

**Comment by knstvk** [comment]
Also, please update after https://github.com/Haulmont/jmix-rest/issues/57 is fixed

File: content/modules/rest/pages/managing-files.adoc (hunk 1 above)
Comment (mariodavid) [reply content/modules/rest/pages/managing-files.adoc]: sounds good. Will do it!

**Pull request merged**